AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m

```diff
@@ -1095,10 +1095,11 @@ - (void)testResumesWhenAllPauseObjectsResumed {
 - (void)testResumeWhenOnlyPausedObjectIsDeallocated {
 
   // If
-  [self.sut pauseWithIdentifyingObjectSync:[NSObject new]];
+  NSObject *object = [NSObject new];
+  [self.sut pauseWithIdentifyingObjectSync:object];
 
   // When
-  [self.sut resumeWithIdentifyingObjectSync:[NSObject new]];
+  [self.sut resumeWithIdentifyingObjectSync:object];
```

Comment:
I don't think this is the right fix. Although this change will make the test check a different valid scenario, the test is specifically for when the "pause" object is deallocated. Maybe instead of calling "resume" we can force the object to be deallocated in the "When" section.

Reply:
Try `object = nil` after pause. If that doesn't work try the whole pause new object inside a block to force switching scope.

Reply:
@achocron I'm not clear how to force the "pause" object to be deallocated in the "When" section. Do you mean remove the object form "pausedIdentifyingObjects" in the "When" section? I pushed a new commit for it, please help to take a look.